### PR TITLE
fix max len issue

### PR DESCRIPTION
**Pull request by cheungdaven**
*Issue #, if available:*

*Description of changes:*

allow user specified max length

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: multimodal/src/autogluon/multimodal/data/process_ner.py [hunk 1]
```diff
@@ -51,12 +52,15 @@ def __init__(
...
         self.prefix = model.prefix
+        self.tokenizer = None
+        self.max_len = max_len
         if self.prefix == NER:
             self.tokenizer = model.tokenizer
-            self.max_len = self.tokenizer.model_max_length
-        else:
-            self.tokenizer = None
-            self.max_len = None
+            if self.tokenizer.model_max_length > max_len:
```
Comment (sxjscience) [inline multimodal/src/autogluon/multimodal/data/process_ner.py:59]: Do we need to check `max_len` against None?

Comment (cheungdaven) [reply multimodal/src/autogluon/multimodal/data/process_ner.py]: fixed. max_len is set to 512 by default in the model config.

**Review by sxjscience — APPROVED** [review]
LGTM

File: multimodal/tests/unittests/test_matcher.py [hunk 2]
```diff
@@ -273,8 +273,8 @@ def test_text_semantic_search():
     for per_query_hits, per_query_hits_2, per_query_hit_gt in zip(hits, hits_2, hits_gt):
         for per_hit, per_hit_2, per_hit_gt in zip(per_query_hits, per_query_hits_2, per_query_hit_gt):
             assert per_hit["response_id"] == per_hit_2["response_id"] == per_hit_gt["response_id"]
-            npt.assert_almost_equal(per_hit["score"], per_hit_2["score"])
-            npt.assert_almost_equal(per_hit["score"], per_hit_gt["score"])
+            npt.assert_allclose(per_hit["score"], per_hit_2["score"], 1e-4, 1e-4)
```
Comment (zhiqiangdon) [inline multimodal/tests/unittests/test_matcher.py:276]: Any idea why 1e-4 is required now? The default precision worked in previous PRs.

**Review by zhiqiangdon — APPROVED** [review]
LGTM! Left a minor comment regarding the precision 1e-4.

File: multimodal/tests/unittests/test_matcher.py (hunk 2 above)
Comment (cheungdaven) [reply multimodal/tests/unittests/test_matcher.py:276]: We encountered a few ci failures from these two lines.

**Review by sxjscience — APPROVED** [review]
LGTM

**Pull request merged**